File: clients/windows/src/Vertex.Shared/TunnelStats.cs
using System.Text.Json.Serialization;

namespace Vertex.Shared;

/// <summary>Cumulative tunnel byte/packet counters since the last connect.</summary>
public sealed record TunnelStats(
    [property: JsonPropertyName("bytesUp")]     long BytesUp     = 0,
    [property: JsonPropertyName("bytesDown")]   long BytesDown   = 0,
    [property: JsonPropertyName("packetsUp")]   long PacketsUp   = 0,
    [property: JsonPropertyName("packetsDown")] long PacketsDown = 0)
{
    public static readonly TunnelStats Zero = new();
}

File: clients/android/Vertex/core/src/main/kotlin/ru/vertices/android/core/ipc/TunnelStats.kt
package ru.vertices.android.core.ipc

import kotlinx.serialization.Serializable

/** Cumulative tunnel byte/packet counters since the last connect. */
@Serializable
data class TunnelStats(
    val bytesUp: Long = 0,
    val bytesDown: Long = 0,
    val packetsUp: Long = 0,
    val packetsDown: Long = 0,
) {
    companion object {
        val ZERO: TunnelStats = TunnelStats()
    }
}
